db/migrate/20201019211215_create_legacy_managers.rb

```diff
@@ -0,0 +1,76 @@
+class CreateLegacyManagers < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
+  include MigrationHelper
+
+  class ExtManagementSystem < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+
+    include MigrationStubHelper
+
+    has_one  :swift_manager,
+             -> { where(:type => "ManageIQ::Providers::StorageManager::SwiftManager" ) },
+             :foreign_key => :parent_ems_id,
+             :class_name  => "CreateLegacyManagers::ExtManagementSystem",
+             :autosave    => true
+
+    has_one  :cinder_manager,
+             -> { where(:type => "ManageIQ::Providers::Openstack::StorageManager::CinderManager" ) },
+             :foreign_key => :parent_ems_id,
+             :class_name  => "CreateLegacyManagers::ExtManagementSystem",
+             :autosave    => true
+
+    def self.openstack
+      where(:type => "ManageIQ::Providers::Openstack::CloudManager")
+    end
+  end
+  
+  class CloudVolume < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class CloudVolumeBackup < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class CloudVolumeSnapshot < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class CloudObjectStoreContainer < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class CloudObjectStoreObject < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    say_with_time("Adding missing Swift and Cinder Managers for OpenStack Cloud Managers") do
+      ExtManagementSystem.openstack.each do |parent|
```

Comment:
Not sure how much this matters since most people won't have 1000s of openstack cloud_managers but we could
`needs_child_cinder_manager = ExtManagementSystem.openstack - ExtManagementSystem.where(:type => "ManageIQ::Providers::Openstack::StorageManager::CinderManager").pluck(:parent_ems_id)`

So that we're not looping over every openstack cloud manager and doing a child query for each.

Might be over-optimization

Reply:
totally don't think this is necessary, but added it

But since this will be in all practical purposes a no-op, good to have minimal data come back